From a4dbda8e6e0bacb33844f0e02527677277167219 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Thu, 12 Jul 2018 17:55:56 +0800
Subject: [PATCH 4149/5242] MLK-18829 mxc IPUv3: vdoa: Replace module_init()
 with subsys_initcall()

commit  1e126efb6f1e43783d310ba390a8a758d8da9ae0 from
https://source.codeaurora.org/external/imx/linux-imx.git

For 4.14 kernel, it turns out that the vdoa driver cannot use
module_init(), otherwise, the driver's probe function cannot be
called.  Using subsys_initcall() makes the probe function be
successfully called, just like the ipu_common driver.  Note that
the mxc_ipu.o object consists of the ipu_common driver and the
vdoa driver.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/ipu3/vdoa.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mxc/ipu3/vdoa.c b/drivers/mxc/ipu3/vdoa.c
index 68194c9..e0128d3 100644
--- a/drivers/mxc/ipu3/vdoa.c
+++ b/drivers/mxc/ipu3/vdoa.c
@@ -1,5 +1,6 @@
 /*
  * Copyright (C) 2012-2015 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2018 NXP
  *
  * The code contained herein is licensed under the GNU General Public
  * License. You may obtain a copy of the GNU General Public License
@@ -528,7 +529,7 @@ static void __exit vdoa_cleanup(void)
 	platform_driver_unregister(&vdoa_driver);
 }
 
-module_init(vdoa_init);
+subsys_initcall(vdoa_init);
 module_exit(vdoa_cleanup);
 
 MODULE_AUTHOR("Freescale Semiconductor, Inc.");
-- 
1.7.9.5

